docker/sawtooth-dev-cxx

```diff
@@ -0,0 +1,86 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+# Description:
+#   Builds an image to be used when developing in Python. The default CMD is to
+#   run build_python.
+#
+#   The image is also used by unit and integration tests. Docker is used to
+#   interact with and orchestrate the creation of other docker containers.
+#
+# Build:
+#   $ cd sawtooth-core/docker
+#   $ docker build . -f sawtooth-dev-python -t sawtooth-dev-python
+#
+# Run:
+#   $ cd sawtooth-core
+#   $ docker run -v $(pwd):/project/sawtooth-core \
+#     -v /var/run/docker.sock:/var/run/docker.sock sawtooth-dev-python
+
+FROM ubuntu:xenial
+
+LABEL "install-type"="mounted"
+
+RUN echo "deb http://repo.sawtooth.me/ubuntu/ci xenial universe" >> /etc/apt/sources.list \
+ && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 6B58B1AC10FB5F63 \
+ && apt-get update \
+ && apt-get install -y -q \
+    apt-transport-https \
+    autoconf  \
+    automake  \
+    build-essential \
+    ca-certificates \
+    cmake  \
+    curl \
+    git \
+    g++  \
+    libcrypto++-dev \
+    liblog4cxx-dev \
+    libtool \
+    libzmqpp-dev \
+    libffi-dev \
+    libssl-dev \
+    make \
+    python3-pip \
+    unzip \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/* \
+ && pip3 install \
+    cpplint
+
+RUN curl -fsSLo protobuf-cpp-3.2.0.zip \
+    https://github.com/google/protobuf/releases/download/v3.2.0/protobuf-cpp-3.2.0.zip \
+ && unzip protobuf-cpp-3.2.0.zip \
+ && cd protobuf-3.2.0 \
+ && ./configure \
+ && make \
+ && make check \
+ && make install \
+ && ldconfig \
+ && cd .. \
+ && rm -rf cd protobuf-3.2.0
```

Comment:
Can this be packaged up as a deb and moved to the package repo so it is just an install? This pushes up the build time substantially for a fresh build.